tests/Unit/Domain/Creators/Test_CylindricalBinaryCompactObject.cpp

```diff
@@ -597,65 +620,114 @@ void test_cylindrical_bbh() {
     const double inner_radius_objectA = 2.0 * mass_ratio / (1.0 + mass_ratio);
     const double inner_radius_objectB = 2.0 / (1.0 + mass_ratio);
 
-    std::unique_ptr<DomainCreator<3>> cyl_binary_compact_object{};
-    if (with_time_dependence) {
-      if (with_boundary_conditions) {
-        cyl_binary_compact_object =
-            TestHelpers::test_option_tag<domain::OptionTags::DomainCreator<3>,
-                                         Metavariables<3, true, true>>(
-                create_option_string(
-                    with_time_dependence,
-                    with_additional_outer_radial_refinement,
-                    with_additional_grid_points, include_outer_sphere,
-                    with_boundary_conditions, use_equiangular_map,
-                    center_objectA, center_objectB, inner_radius_objectA,
-                    inner_radius_objectB, outer_radius));
-      } else {
-        cyl_binary_compact_object =
-            TestHelpers::test_option_tag<domain::OptionTags::DomainCreator<3>,
-                                         Metavariables<3, true, false>>(
-                create_option_string(
-                    with_time_dependence,
-                    with_additional_outer_radial_refinement,
-                    with_additional_grid_points, include_outer_sphere,
-                    with_boundary_conditions, use_equiangular_map,
-                    center_objectA, center_objectB, inner_radius_objectA,
-                    inner_radius_objectB, outer_radius));
-      }
+    CylBCO::InitialRefinement::type initial_refinement{};
+    CylBCO::InitialGridPoints::type initial_grid_points{};
+
+    const auto make_initial_structure =
+        // https://www.open-std.org/jtc1/sc22/wg21/docs/papers/2017/p0588r1.html
+        // ```
+        // If a lambda-expression [...] captures a structured binding
+        // (explicitly or implicitly), the program is ill-formed.
+        // ```
+        // There is a bug in the standard where structured bindings cannot by
+        // captured in a lambda. This is why we pass them in as arguments here
+        [](const size_t initial_value, const bool with_inner_A,
+           const bool with_inner_B, const bool with_outer_sphere) {
```

Comment:
Move this lambda to a separate function, that's more readable anyway.

---

FYI: This is an annoying behavior of the standard, I agree. You can work around it by capturing the variables by name and by value like this:

```cpp
const auto lambda = [include_inner_sphere_A = include_inner_sphere_A, ...]() { ... };
```

You can even use the same names I believe.